### PR TITLE
docs(web modeler): add feature documentation for git sync

### PR DESCRIPTION
**Pull request by conceptualshark**
## Description

Resolves https://github.com/camunda/web-modeler/issues/9706, feature documentation for Git Sync (GitHub only).

Related to https://github.com/camunda/product-hub/issues/2036

## When should this change go live?

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**.
- [ ] This is already available but undocumented and should be released within a week.
- [ ] This on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (apply `hold` label or convert to draft PR)
- [x] This is part of a scheduled alpha or minor. (apply alpha or minor label)
- [ ] There is **no urgency** with this change and can be released at any time.

## PR Checklist

- [ ] My changes are for an **already released minor** and are in `/versioned_docs` directory.
- [x] My changes are for the **next minor** and are in `/docs` directory (aka `/next/`).

- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned the Engineering DRI or delegate.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @camunda/tech-writers as a reviewer.

### CODE REVIEW

**Comment by marcellobarile** [comment]
Adding @JacobiCamunda so he can give his BE engineer point of view regarding possible errors and known limitations.

**Review by JacobiCamunda — CHANGES_REQUESTED** [review]
Very nice to read! 🚀 
I reviewed from the backend perspective for the current implementation. I will add another review if necessary when the push operation is fully integrated.

CC: @marcellobarile

File: docs/components/modeler/web-modeler/git-sync.md
Comment (marcellobarile) [inline docs/components/modeler/web-modeler/git-sync.md:54]:
```suggestion
Organization owners/administrators, project administrators, and project editors can sync their version of Web Modeler with the connected GitHub repository at any time.
```

File: docs/components/modeler/web-modeler/git-sync.md
Comment (marcellobarile) [inline docs/components/modeler/web-modeler/git-sync.md:70]:
```suggestion
Existing GitHub configurations can be edited from the gear icon beside the **Sync with GitHub** button. Permission to update these settings are limited by the roles within your organization and project.
```

File: docs/components/modeler/web-modeler/git-sync.md
Comment (marcellobarile) [inline docs/components/modeler/web-modeler/git-sync.md:74]:
```suggestion
- **Project administrators - SaaS:** Edit and update only the **GitHub repository URL**, and **branch name**.
```

**Review by marcellobarile — APPROVED** [review]
🔝 very nicely done! Thanks!

File: docs/components/modeler/web-modeler/git-sync.md
Comment (conceptualshark) [reply docs/components/modeler/web-modeler/git-sync.md]: I went ahead and refactored this section to include a variation of the above, and added a note regarding main process assignment to the initial sync section as well - let me know if that works to cover this.

File: docs/components/modeler/web-modeler/git-sync.md
Comment (marcellobarile) [inline docs/components/modeler/web-modeler/git-sync.md:46]: 💭 maybe we could specify even more what `Web Modeler has assigned the correct main process` means? @JacobiCamunda any thoughts on how to improve it even more? I remember you verbalized this case in some past discussion.

Comment (JacobiCamunda) [reply docs/components/modeler/web-modeler/git-sync.md:46]: My verbalization was incorporated in the "File name conventions" part at the end of the document. I think we can keep it this way 👍

Comment (JacobiCamunda) [reply docs/components/modeler/web-modeler/git-sync.md:46]: But I think we should mention the initial sync and how it works a bit differently + it is done always when branch switching. What do you think, @marcellobarile?

**Comment by marcellobarile** [comment]
@conceptualshark there is a small change in the settings dialog;
We are replacing `Application ID` and `Owner name` with `Installation ID` and `Client ID`.
The `Installation ID` can be retrieved as explained [here](https://docs.retool.com/source-control/quickstarts/setup/github#install-the-github-app)

The new UI: 
image

The issue for this change:
https://github.com/camunda/web-modeler/issues/10302

Excuse us for the last-minute change :) I hope that's fine.

FYI @JacobiCamunda

File: docs/components/modeler/web-modeler/git-sync.md
Comment (christinaausley) [inline docs/components/modeler/web-modeler/git-sync.md:4]: Not blocking by any means, but this might flag Bing as too short -- I would detail. For example, "Sync your Git repositories with Web Modeler to achieve X, Y, and Z."

File: docs/components/modeler/web-modeler/git-sync.md
Comment (christinaausley) [inline docs/components/modeler/web-modeler/git-sync.md:49]: A good opportunity to add a screenshot.

File: docs/components/modeler/web-modeler/git-sync.md
Comment (christinaausley) [inline docs/components/modeler/web-modeler/git-sync.md:59]: Is "Github" what it reads in the UI? Should adjust to GitHub.

File: sidebars.js
Comment (christinaausley) [inline sidebars.js:151]: Is this the preferred placement/developer journey by the Modeler team?

Comment (marcellobarile) [reply sidebars.js:151]: That was the one that we agreed on.

**Review by christinaausley — APPROVED** [review]

File: docs/components/modeler/web-modeler/git-sync.md
Comment (marcellobarile) [inline docs/components/modeler/web-modeler/git-sync.md:52]: 💭 previously we said:
```
Once redirected to your application's installation page, copy the **Installation ID** located at the end of the page's URL: `https://github.com/settings/installations/{installation_id}`.
```

Maybe we should refer to this; I'm not sure that the `Installation ID` is prominent on the settings page.

File: docs/components/modeler/web-modeler/git-sync.md
Comment (marcellobarile) [inline docs/components/modeler/web-modeler/git-sync.md:54]: 💭 also here, we should probably mention that the downloaded PEM file can be opened with a text editor, and its content can be copy/pasted

**Comment by JacobiCamunda** [comment]
@marcellobarile, a part would be good to say that actions which alters the commit sha which Web Modeler is currently synced to (like a squash) in the remote branch could break the git sync.

**Comment by marcellobarile** [comment]
@conceptualshark this morning we have acknowledged that timeouts during the sync operations might occur (we don't know how often yet). Is it worth mentioning this in the known limitations? the only thing that the user could to so far is retrying.

**Review by marcellobarile — APPROVED** [review]
👍

**Comment by conceptualshark** [comment]
@marcellobarile I believe we're good to go - just let me know if it can be merged now (and if you'd like a docs release at any point). Otherwise, I can put a `hold` label on it until you're ready.

**Comment by marcellobarile** [comment]
@conceptualshark thanks for the update! It's good to know that the doc is ready, just put it on hold because we are still about to kick-off the QA.
Let's merge this in the context of the next release. I'll keep you posted when ready.

**Comment by akeller** [comment]
> @conceptualshark thanks for the update! It's good to know that the doc is ready, just put it on hold because we are still about to kick-off the QA. Let's merge this in the context of the next release. I'll keep you posted when ready.

Popping in to say I **_LOVE_** that the docs were drafted ahead of QA! Having them available means QA can provide feedback on the docs experience as part of the overall epic experience. Great job! 🧡

**Review by JacobiCamunda — APPROVED** [review]

**Comment by marcellobarile** [comment]
@conceptualshark the feature has been merged to the main branch and will be released on SaaS tomorrow. Ping me once your release is done so we can orchestrate ours.

**Comment by conceptualshark** [comment]
@marcellobarile The docs should now be live: https://docs.camunda.io/docs/next/components/modeler/web-modeler/git-sync/

**Pull request merged**
